File: Alay/Sliding-Window/1248.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int count = 0, oddCount = 0, left = 0, right = 0;
        int n = nums.size();
        int prefixCount = 0;

        while (right < n) {
            if (nums[right] % 2 == 1) {
                oddCount++;
                prefixCount = 0;
            }
            while (oddCount == k) {
                if (nums[left] % 2 == 1) {
                    oddCount--;
                }
                prefixCount++;
                left++;
            }
            count += prefixCount;
            right++;
        }

        return count;
    }
};

File: Aditya-Pratap/Sliding Window/1248.py
class Solution:
    def subarray(self,nums:List[int],k:int)->int:
        count,odd=0,0
        i=0
        for j in range(len(nums)):
            if nums[j]%2==1:
                odd+=1
            while odd>k:
                if nums[i]%2==1:
                    odd-=1
                i+=1
            count+=j-i+1
        return count
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        return self.subarray(nums,k)-self.subarray(nums,k-1)
        